src/context/AuthReducer.tsx

```
@@ -0,0 +1,37 @@
+
+interface AuthAction {
+  type: string;
+  payload?: string | number | boolean | object | null;
+}
+
+const AuthReducer = (state: AuthState, action: AuthAction): AuthState => {
+    switch (action.type) {
+      case "LOGIN": {
+        // Validate payload has expected User properties before casting
+        const isValidUser = action.payload && 
+          typeof (action.payload as any).email === 'string';
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)